common/OpTestUtil.py

```diff
@@ -2347,6 +2347,42 @@ def prepare_source(self,spec_file, host, dest_path, package, build_option=None):
             return host.host_run_command(f"ls -d -- {dest_path}/* | grep {package}")[0]
         except OpTestError:
             return ""
+        
+    def set_attribute_redfish(self, uri, attribute_name, attribute_value):
+        """
+        Changing any attribute value using Redfish API
+        
+        :param uri: redfish uri at which the attribute can be updated
+        :param attribute_name: Should be same as attribute name in redfish
+        :param attribute_value: Value want be be updated for attribute
+        """
+        auth_token = self.generate_ssl_auth_token(ip_add=self.conf.args.bmc_ip)
+        content_type = "-H 'Content-Type: application/json'"
+        rest_server = "https://{}/{}".format(self.conf.args.bmc_ip, uri)
+        attribute_param = '{"Attributes":{{attribute_name}:{attribute_value}}}'
+        curl_command = "curl -k -H X-Auth-Token: "+auth_token+" "+content_type+f" -X PATCH {rest_server} "+f"-d {attribute_param}"
+        try:
+            output = self.run_command(term_obj, curl_command)
+        except CommandFailed as cf:
+            return cf.output
+        return output
+        
+    def generate_ssl_auth_token(self, ip_add = None):
+        """
+        Generates ssl key and returns the ssl key
+        """
```

Comment:
i feel both of these function moved in `common/OpTestEBMC.py` it well helps to tight coupling as these will never used outside EBMC use case

Reply:
Hi Praveen! These definitions can be used across all Redfish interfaces. I hope it's not limited only to EBMC